lib/BackbeatProducer.js

```diff
@@ -57,7 +57,7 @@ class BackbeatProducer extends EventEmitter {
                 pollIntervalMs: joi.number().default(DEFAULT_POLL_INTERVAL),
                 maxRequestSize: joi.number().default(KAFKA_PRODUCER_MESSAGE_MAX_BYTES),
                 compressionType: joi.string().default(KAFKA_PRODUCER_DEFAULT_COMPRESSION_TYPE),
-                requiredAcks: joi.number().default(KAFKA_PRODUCER_DEFAULT_REQUIRED_ACKS),
+                requiredAcks: joi.string().default(KAFKA_PRODUCER_DEFAULT_REQUIRED_ACKS),
```

Comment:
why pass a string? this reduces the effectiveness of validation, as any string will be valid...

if we just neet to support some way of passing "empty" param to kafka, i.e. use kafka defaults, we should be able to just do `joi.number().allow("").default(KAFKA_PRODUCER_DEFAULT_REQUIRED_ACKS)` : which (according to the docs) should allow numbers or empty string, and fall back to default value if `undefined` value is provided

Reply:
I passed a string because the default value used in Artesca is 'all' which waits for all brokers to ack. Looks like we can use -1 too for the same purpose, Leaving it as a number.